src/liiteri/mime.clj

```diff
@@ -14,3 +17,16 @@
                                                :allowed-content-types allowed-mime-types})))
       real-content-type)))
 
+(defn fix-extension [filename real-content-type]
+  (let [extension-from-mimetype (mime/extension-for-name real-content-type)
+        [fname ext] (fs/split-ext filename)]
+    (format "%s%s" fname extension-from-mimetype)))
```

Comment:
Kummasta osassa piste on mukana? Miten käy jos alkuperäisessä tiedostonimessä ei ollut pistettä?

Reply:
`(fs/split-ext filename)` poistaa alkuperäisessä tiedostossa olevat pisteet, ja `(mime/extension-for-name real-content-type)` palauttaa aina pisteellisen liitteen. Lisäsin tälle testin.